framework/src/main/java/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1204,6 +1185,18 @@ private void annotate(AnnotatedTypeMirror type, List<AnnotationExpr> annotations
         }
         for (AnnotationExpr annotation : annotations) {
             AnnotationMirror annoMirror = getAnnotation(annotation, allStubAnnotations);
+            if (annoMirror == null) {
+                // If the annotation was not imported, then #getAllStubAnnotations does
+                // not add it to the allStubAnnotations field. This code compensates for
+                // that deficiency by adding the annotation when it is encountered (i.e. here).
+                TypeElement annoTypeElt = getTypeElement(annotation.getNameAsString());
```

Comment:
Just call `TypeElement classElement = elements.getTypeElement(typeName);` directly so that the more sensible warning will be issued below.  (Make sure to document why getTypeElement isn't called.)